src/CommunityToolkit/Xamarin.CommunityToolkit/Views/DockLayout.shared.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    public enum Dock
+    {
+        Left,
+        Top,
+        Right,
+        Bottom
+    }
+
+    /// <summary>
+    /// The DockLayout makes it easy to dock content in all four directions (top, bottom, left and right).
+    /// This makes it a great choice in many situations, where you want to divide the screen into specific areas,
+    /// especially because by default, the last element inside the DockLayout, unless this feature is specifically disabled,
+    /// will automatically fill the rest of the space (center).
+    /// Inspired by WPF DockPanel: https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.dockpanel?view=netframework-4.8
+    /// </summary>
+    public class DockLayout : Layout<View>
+    {
+        public static readonly BindableProperty DockProperty =
+            BindableProperty.Create(nameof(Dock), typeof(Dock), typeof(DockLayout), Dock.Left,
+                BindingMode.TwoWay, null);
+
+        public Dock Dock
+        {
+            get => (Dock)GetValue(DockProperty);
+            set => SetValue(DockProperty, value);
+        }
+
+        Dock GetDock(BindableObject bindable) => (Dock)bindable.GetValue(DockProperty);
```

Comment:
Shouldn't we replace it by **CreateAttached** and two **static** methods Get/Set ?

Reply:
Good note, changed.